### DISCUSSION TITLE
Tiled textures have seams where they repeat

### DISCUSSION THREAD

**Discussion by Jade-Harleyy** [FakeFishGames/Barotrauma#13920] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Despite the sourcerects being correct, this bug still occurs:
![image](https://github.com/FakeFishGames/Barotrauma/assets/67431770/054f2565-58e4-4386-bd7d-a0b57cc9485d)

### Reproduction steps

1. Download and enable this mod: [More Decoratives.zip](https://github.com/FakeFishGames/Barotrauma/files/15196004/More.Decoratives.zip)
2. Open the sub editor and place down the "Large Inside Wall" structure.
3. Extend the structure.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by BlackMoons** [comment]
this is because baro issues seperate geometry for every tile of a texture, subpixel errors are bound to occur on certain hardware.

Proper way to fix would be using the pixel shader to do the UV repeat, and manually issuing DDX/DDY values to the sample lookup in the shader to prevent the weird mipmap choice at UV wrapping point.

Try turning off FSAA if you have it force enabled in your video drivers (or any other 'enhancements)

And note that the errors are generally worse in sub editor then they are in game (or even in test mode)

Textures may also need a pixel or two of bleed around them to prevent dark edges where it repeats, since mipmaps (and just regular texture sampling) will sample adjacent pixels to the ones you actually specify.
